### PR TITLE
Add kubernetes.container.status.last.reason metric

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Signed-off-by: Tetiana Kravchenko <tetiana.kravchenko@elastic.co>

## What does this PR do?

Introduce new metric `kubernetes.container.status.last.reason`

## Why is it important?

we do already have `kubernetes.container.status.reason` (that relies on `kube_pod_container_status_terminated_reason` and `kube_pod_container_status_waiting_reason` - https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/state_container/state_container.go#L74-L75 ), but this one will only be reported when the container/pod is actually down. This might be a quite short period of time ( this is the initial reason of adding `kube_pod_container_status_last_terminated_reason` - https://github.com/kubernetes/kube-state-metrics/pull/535)

I did not map `kube_pod_container_status_last_terminated_reason` to `kubernetes.container.status.reason`, because existence of `kube_pod_container_status_last_terminated_reason` do not exclude `kube_pod_container_status_waiting_reason` or `kube_pod_container_status_terminated_reason` ( for  `kube_pod_container_status_waiting_reason` and `kube_pod_container_status_terminated_reason` it is fine to be maped to the same `kubernetes.container.status.reason`, because those stated exclude each other - https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-states)

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- closes https://github.com/elastic/beats/issues/27840

## Use cases

described in https://github.com/elastic/beats/issues/27840:
```
 In incidents the customer experiences, the pod is restarting so fast, there is only a little chance to capture this event.
```

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -180,6 +180,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 - Add `xpack.enabled` support for Enterprise Search module. {pull}29871[29871]
 - Add gcp firestore metricset. {pull}29918[29918]
 - Remove strict parsing on RabbitMQ module {pull}30090[30090]
+- Add `kubernetes.container.status.last.reason` metric {issue}27840|[27840]
```
Comment (MichaelKatsoulis) [inline CHANGELOG.next.asciidoc:183]:
```suggestion
- Add `kubernetes.container.status.last.reason` metric {issue}30306|[30306]
```

**Review by ChrsMark — COMMENTED** [review]
I would go with sth like `kubernetes.container.status.terminated_reason`.

**Comment by tetianakravchenko** [comment]
@ChrsMark  `kubernetes.container.status.terminated_reason` might be confusing and potentially mixed up with metric `kube_pod_container_status_terminated_reason`, that describes the reason the pod is currently in terminated state. 

The most precise naming would be `kubernetes.container.status.last.terminated_reason` as @MichaelKatsoulis suggested, or maybe  `kubernetes.container.status.last_terminated_reason` if we want to avoid additional layer of nested object. Let me know what do you think about it.

**Comment by ChrsMark** [comment]
+1 for `kubernetes.container.status.last_terminated_reason`

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (tetianakravchenko) [reply CHANGELOG.next.asciidoc]: the issue number is 27840, but anyway - changed it to `pull` https://github.com/elastic/beats/pull/30306/commits/28f2f7ec2130b8127fd429126a699fa769874290

**Review by ChrsMark — APPROVED** [review]
🚀

**Pull request merged**